src/constants/triggers.ts

```diff
@@ -0,0 +1,89 @@
+export enum ResourceType {
+  TASK = "TASK",
+  VERSION = "VERSION",
+  BUILD = "BUILD",
+}
+
+type PayloadResourceIdKey = "in-version" | "in-build" | "id";
+export interface Trigger {
+  trigger: string;
+  label: string;
+  extraFields?: ExtraField[];
+  resourceType: ResourceType;
+  payloadResourceIdKey?: PayloadResourceIdKey;
+  regexSelectors?: RegexSelector[];
+}
+
+export const buildRegexSelectors: RegexSelector[] = [
+  {
+    type: "display-name",
+    typeLabel: "Build Variant Name",
+  },
+  {
+    type: "build-variant",
+    typeLabel: "Build Variant ID",
+  },
+];
+
+export const taskRegexSelectors: RegexSelector[] = [
+  {
+    type: "display-name",
+    typeLabel: "Task Name",
+  },
+  {
+    type: "build-variant",
+    typeLabel: "Build Variant Name",
+  },
+];
+
+export interface ExtraField {
+  text: string;
+  key: string;
+  type?: string;
+  options?: StringMap;
+  default?: string;
+  validator?: (v: any) => string;
+  dataCy?: string;
+}
+
+export type RegexSelectorType = "display-name" | "build-variant";
+export interface RegexSelector {
+  type: RegexSelectorType;
+  typeLabel: string;
+}
+
+export const failureTypeSubscriberConfig = {
+  text: "Failure type",
```

Comment:
In some places we use capitalization across (ex: Build Variant Name) but not consistently. I'm sure this was a failure of the legacy UI but I think it would be good to make this consistent now -- personally I prefer capitalizing (Failure Type) but whatever works

Reply:
done